Userland/Libraries/LibWeb/Fetch/Infrastructure/HTTP/Requests.cpp

```diff
@@ -364,12 +364,9 @@ bool Request::cross_origin_embedder_policy_allows_credentials() const
     // FIXME: 3. If request’s client’s policy container’s embedder policy’s value is not "credentialless", then return true.
 
     // 4. If request’s origin is same origin with request’s current URL’s origin and request does not have a redirect-tainted origin, then return true.
-    // FIXME: Actually use the given origins once we have https://url.spec.whatwg.org/#concept-url-origin.
-    if (HTML::Origin().is_same_origin(HTML::Origin()) && !has_redirect_tainted_origin())
-        return true;
-
     // 5. Return false.
-    return false;
+    auto const* request_origin = m_origin.get_pointer<HTML::Origin>();
+    return request_origin->is_same_origin(URL::url_origin(current_url())) && !has_redirect_tainted_origin();
```

Comment:
We can't assume `m_origin` has `HTML::Origin` - if it has `Origin::Client`, this will crash. Look at how `has_redirect_tainted_origin` checks if this pointer is null.

Reply:
Yeah had a feeling that was going to be a problem, was just confused with what the spec expected vs the `Variant<>` usage. I'll go ahead and just check if `request_origin` is null and return false as we can't do anything useful with the enum class variant.